Loop times revamp

PikaDude opened the pull request:
This PR introduces the following:
- An upgraded loop times module that now supports weather
- A loop times debug function, that will skim the song to right before the loop. Used to test the loop
- Loop times for every single hourly song (all 192). These loop times now allow for seamless song looping, skipping any intros or outros
- A fix causing the loop timer to start before the song actually begins playing
- Optimisations and fixes that don't make the loop timer reload the song every loop
- A small fix for the `formatHour` function, making the checks non-strict, so passing strings will now result in an accurate response

PikaDude committed:
Update: revamp looptimes functionality

PikaDude committed:
Fix: remove testing line

PikaDude committed:
Fix: error if song has no loop times

PikaDude committed:
Add: animal crossing sunny loop times

PikaDude committed:
Add: animal crossing snowing loop times

I copied all of the loop times from the sunny
variation. I haven't tested every single song,
but it appears that these loop times are
working as well for the snowing variation of
the songs.

PikaDude committed:
Fix: make formatHour comparisons non-strict

PikaDude committed:
Add: wild world sunny loop times

PikaDude committed:
Add: wild world snowing loop times

Copied over from the sunny variation, and I've
tested each song. Any issues have been corrected.

PikaDude committed:
Add: wild world raining loop times

Copied over from the sunny variation, and I've
tested each song. Any issues have been corrected.

PikaDude committed:
Fix: wasn't meant to push those (whoops)

PikaDude committed:
Add: new leaf sunny loop times

PikaDude committed:
Add: new leaf snowing loop times

PikaDude committed:
Add: new leaf raining loop times

PikaDude committed:
Update: loop times comment

JordiGarcL commented:
I've been postponing reviewing this PR as it would require some time. I should be able to allocate time tomorrow!

PikaDude committed:
Merge branch 'develop' into feature/loop-times-revamp

PikaDude committed:
Fix: looptimes incompatibility w/ music pausing

This important commit now allows us to safely
pause the music without ruining the looptimes.

PikaDude committed:
Remove: this part was not necessary

JordiGarcL commented:
Will check this now

jordigarcl commented on scripts/background/Utility.js:
```diff
@@ -6,13 +6,13 @@ var DEBUG_FLAG = false;
 // format text for the badge and for the song URL
 function formatHour(time) {
-	if (time === -1) {
+	if (time == -1) {
```
What's the reason for this change? I guess we are sending `time` as a string somewhere?  Damn javascript and its equality checks

PikaDude replied:
This was an issue that occurred during testing, it's not really needed, however was done to prevent any future issues as strict equality checks should only be used when necessary.
